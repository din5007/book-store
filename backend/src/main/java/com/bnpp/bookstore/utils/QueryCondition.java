package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
