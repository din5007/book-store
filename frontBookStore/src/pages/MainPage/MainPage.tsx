
import { FC } from "react";
import Container from "react-bootstrap/esm/Container";
import Row from "react-bootstrap/esm/Row";
import "./MainPage.css";

const MainScreen : FC<{ title : string, children : any}> = ({ title, children }) => {
  return (
    <div className="mainback">
      <Container>
        <Row>
          <div className="page">
            {title && (
              <>
                <h1 className="heading">{title}</h1>
              </>
            )}
            {children}
          </div>
        </Row>
      </Container>
    </div>
  );
};

export default MainScreen;
