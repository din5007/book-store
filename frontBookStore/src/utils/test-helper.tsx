import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;