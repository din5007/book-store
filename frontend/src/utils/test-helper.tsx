import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;