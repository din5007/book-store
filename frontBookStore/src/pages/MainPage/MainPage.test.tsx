import { customRender as render } from '../../utils/test-helper';
import MainPage from "./MainPage"
import { describe, expect, it } from 'vitest';

describe('Main Page', () => {
    it('Main Page should load properly - snapshot', async () => {
        const {container}  = render(<MainPage title={'Login'} children={<p> child component goes here </p>} />);
        expect(container).toMatchSnapshot();
    });
})