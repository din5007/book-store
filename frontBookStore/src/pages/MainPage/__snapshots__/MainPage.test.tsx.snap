// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main Page > Main Page should load properly - snapshot 1`] = `
<div>
  <div
    style="min-height: 93vh; display: flex; padding: 10px 0px;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="w-100"
        >
          <h1>
            Login
          </h1>
          <p>
             child component goes here 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
